src/core.js

```diff
@@ -822,7 +823,7 @@ $.extend( $.validator, {
 					if ( group ) {
 						$.each( this.groups, function( name, testgroup ) {
 							if ( testgroup === group ) {
-								$( "[name='" + name.replace( /'/g, "\\'" ) + "']", this.currentForm )
+								$( "[name='" + this.escapeCssMeta( name ) + "']", this.currentForm )
```

Comment:
I think It's my fault, as I forgot that the `this` inside `$.each( ... )` refer to the current value of the `object|array` we iterate over.

I'll investigate the issue and put together a PR to fix it with a unit test ASAP